plc4j/drivers/firmata/src/main/java/org/apache/plc4x/java/firmata/readwrite/protocol/FirmataProtocolLogic.java

```diff
@@ -71,12 +71,22 @@ public void onConnect(ConversationContext<FirmataMessage> context) {
         LOGGER.debug("Sending Firmata Reset Command");
         FirmataMessageCommand resetCommandMessage = new FirmataMessageCommand(new FirmataCommandSystemReset());
         context.sendRequest(resetCommandMessage)
+            .onTimeout(e -> {
+                LOGGER.info("Timeout during Connection establishment, closing channel...");
```

Comment:
is the closing implicit or is there missing a close call here?

Reply:
Öhm ... from where I copied it from, it was commented out ;-)

Reply:
Maybe @splatch added that? :D. Anyway maybe it is better to name it `channel will be closed` as there is no immediate close call there.

Reply:
The `onTimeout` is not propagated well across stack, from what I remember in the past `onTimeout` had to specified at first, otherwise it was ignored/swallowed by lower part of engine. Not sure if that's true nowadays.